File: test-data/typescript/circular_dependency/moduleA.ts
import { ClassB } from './moduleB';

export class ClassA {
    private b: ClassB | undefined;

    constructor(public name: string) { }

    setB(b: ClassB) {
        this.b = b;
    }

    callB() {
        if (this.b) {
            return this.b.methodB();
        }
        return "No B set";
    }
}

File: test-data/python/circular_dependency/module_a.py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .module_b import ClassB

class ClassA:
    def __init__(self, name: str):
        self.name = name
        self.b: 'ClassB' = None

    def set_b(self, b: 'ClassB'):
        self.b = b

    def call_b(self):
        if self.b:
            return self.b.method_b()
        return "No B set"
